src/main/java/nl/ordina/jobcrawler/controller/JobcrawlerController.java

```diff
@@ -1,73 +1,95 @@
+import java.util.UUID;
 
 
 @CrossOrigin(origins = "http://localhost:4200")
 @RestController
 public class JobcrawlerController {
 
     private VacancyService vacancyService;
+    private SkillService skillService;
 
     @Autowired
-    public JobcrawlerController(VacancyService vacancyService) {
+    public JobcrawlerController(VacancyService vacancyService, SkillService skillService) {
         this.vacancyService = vacancyService;
+        this.skillService = skillService;
     }
 
-    // Post mappings
-
     @PostMapping("/searchrequest")
     public SearchResult searchRequest(@RequestBody SearchRequest request) {
-        return new SearchResult(request, vacancyService.getAllJobs());
+        return new SearchResult(request, vacancyService.getAllVacancies());
     }
 
+
+    //******** Adding ********//
     @PostMapping("/addJobWithJson")
     public Vacancy addJob(@Valid @RequestBody Vacancy job) {
         return vacancyService.add(job);
     }
 
-    // Get mappings
 
+    //******** Getting ********//
     @GetMapping("/getByID/{id}")
     public Vacancy getByID(@PathVariable("id") UUID id) {
         // Retrieve a vacancy by its ID (UUID). If Vacancy is not found it throws a 'VacancyNotFoundException' (HttpStatus.NOT_FOUND).
-        Optional<Vacancy> vacancy = vacancyService.getByID(id);
+        Optional<Vacancy> vacancy = vacancyService.getVacancyByID(id);
         return vacancy.orElseThrow(() -> new VacancyNotFoundException("Vacancy with id: " + id + " not found."));
     }
 
     @GetMapping("/getJobsByBroker/{broker}")
     public List<Vacancy> getJobsByBroker(@PathVariable("broker") String broker) {
         // Retrieve all vacancies from a specific broker. Currently case sensitive
-        return vacancyService.getJobsByBroker(broker);
+        return vacancyService.getVacanciesByBroker(broker);
     }
 
     @GetMapping("/getAllJobs")
     public List<Vacancy> getAllJobs() {
         // Retrieve all vacancies that are available in the database
-        return vacancyService.getAllJobs();
+        return vacancyService.getAllVacancies();
+    }
+
+    @GetMapping(path = "skills")
+    public List<Skill> getAllSkills() {
+        return skillService.getAllSkills();
     }
 
     @GetMapping("/getJobsWithSkill/{skill}")
-    public List<Vacancy> getJobsWithSkill(@PathVariable("skill") String skill) {
+    public Set<Vacancy> getJobsWithSkill(@PathVariable("skill") String skill) {
         // Only show vacancies which needs a specific skill that's requested via a get method
-        return vacancyService.getJobsWithSkill(skill);
+        return vacancyService.getVacanciesBySkill(skill);
     }
 
-    // Delete mappings
 
+    //******** Deleting ********//
     @DeleteMapping("/delete/{id}")
-    public void deleteAanvraagById(@PathVariable("id") UUID id) {
+    public void deleteVacancyById(@PathVariable("id") UUID id) {
         // Delete vacancy by id (UUID)
         vacancyService.delete(id);
     }
 
+
```

Comment:
What does the update do exactly? 
Is the list of skills updated as well?

Reply:
done